misc.inc.php

```diff
@@ -868,7 +868,9 @@ function buildnavmenu($ma,&$tl){
 */
 
 if( AUTHENTICATION=="Saml" && !isset($_SESSION['userid']) && php_sapi_name()!="cli" ){
-	header("Location: ".redirect('saml/login.php'));
+	$savedurl = $_SERVER['SCRIPT_NAME'] . "?" . sanitize($_SERVER['QUERY_STRING']);
+	setcookie( 'targeturl', $savedurl, time()+60 );
+	header("Location: ".redirect($config->ParameterArray["InstallURL"].'/saml/login.php'));
```

Comment:
This assumes that the InstallURL parameter is set, we already have logic to deal with that in the redirect function.  https://github.com/samilliken/openDCIM/blob/master/misc.inc.php#L132 so you should just need the relative location here and not attempt to set the install url.

`	if($installURL!=""){
		// $installURL isn't blank so combine it with the target to get a valid redirect target
		$url = $installURL.$target;`

Reply:
I have reverted that and found that the issue that I was trying to solve is fixed by sanitizing the REQUEST_URI in the path function.  I am not sure if that is okay in this situation as it may cause issues i am not aware of.

Essentially I wanted to check if there would be an issue with redirecting to the URL that the user was attempting to access on the redirect to the saml page.  I kept getting a server error and a long invalid URI:

https://localhost/devices.php?DeviceID=1831&[SNIP]/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/saml/login.php